cdap-app-fabric/src/main/java/io/cdap/cdap/gateway/handlers/OperationHttpHandler.java

```diff
@@ -98,4 +156,76 @@ public void failOperation(FullHttpRequest request, HttpResponder responder,
         String.format("Updated status for operation run %s in namespace '%s'.", runId,
             namespaceId));
   }
+
+  private ScanOperationRunsRequest getScanRequest(
+      String namespaceId, String pageToken, Integer pageSize, String filter) {
+    ScanOperationRunsRequest.Builder builder = ScanOperationRunsRequest.builder();
+    builder.setNamespace(namespaceId);
+    if (pageSize != null) {
+      builder.setLimit(pageSize);
+    }
+    if (pageToken != null) {
+      builder.setScanAfter(pageToken);
+    }
+    if (filter != null && !filter.isEmpty()) {
+      OperationRunFilter operationRunFilter = getFilter(filter);
+      builder.setFilter(operationRunFilter);
+    }
+    return builder.build();
+  }
+
+  private OperationRunFilter getFilter(String filter) {
+    Map<String, String> filterKeyValMap = parseFilter(filter);
+    OperationType operationType = null;
+    OperationRunStatus operationStatus = null;
+
+    for (Map.Entry<String, String> entry : filterKeyValMap.entrySet()) {
+      String key = entry.getKey();
+      String value = entry.getValue();
+
+      switch (key) {
+        case "type":
```

Comment:
We should add a enum for this

OpreationFilterKey{
TYPE
STATUS
}

We convert the `key` string to the enum. After that we can use switch case

Reply:
Yes right. Thank you for the correction